### PR TITLE
nrf_oberon v3.0.8

### PR DESCRIPTION
**Pull request by frkv**
-Adding library nrf_oberon v3.0.8
-Adding incremental ECDH for secp256r1
-Adding ocrypto_ APIs for SHA-224 and SHA-384
-Adding ocrypto_ APIs for pbkdf2 (SHA-1 and SHA-256)
-Updating CHANGELOG.rst

manifest pr: https://github.com/nrfconnect/sdk-nrf/pull/3778

ref: NCSDK-7780

Signed-off-by: Frank Audun Kvamtrø <frank.kvamtro@nordicsemi.no>

### CODE REVIEW

File: crypto/nrf_oberon/include/ocrypto_hmac_sha256.h [hunk 1]
```diff
@@ -132,6 +132,23 @@ void ocrypto_hmac_sha256(
     const uint8_t* key, size_t key_len,
     const uint8_t* in, size_t in_len);
 
+/**
+ * HMAC-SHA256 algorithm with AAD.
+ *
+ * @param[out] r       HMAC output
```
Comment (Vge0rge) [inline crypto/nrf_oberon/include/ocrypto_hmac_sha256.h:138]:
```suggestion
 * @param[out] r       HMAC output.
```

File: crypto/nrf_oberon/include/ocrypto_pbkdf2.h [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+/**@file
+ * @defgroup nrf_oberon_pbkdf2 PBKDF2 APIs
+ * @ingroup nrf_oberon
+ * @{
+ * @brief Type declaration and APIs for PBKDF2 with HMAC-SHA1.
+ *
+ * PBKDF2 with HMAC-SHA1 is password-based key derivation function defined in RFC2898.
```
Comment (Vge0rge) [inline crypto/nrf_oberon/include/ocrypto_pbkdf2.h:13]:
```suggestion
 * PBKDF2 with HMAC-SHA1 is a password-based key derivation function defined in RFC2898.
```

**Review by Vge0rge — APPROVED** [review]

File: crypto/nrf_oberon/include/ocrypto_curve_p224.h [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+#define OCRYPTO_CURVE_P224_H
+
+#include "ocrypto_sc_p224.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+// (x,y) only jacobian coordinates
```
Comment (torsteingrindvik) [inline crypto/nrf_oberon/include/ocrypto_curve_p224.h:25]: Should all the instances of `jacobian` perhaps be capitalized as `Jacobian`?

File: crypto/nrf_oberon/include/ocrypto_curve_p224.h [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+ *
+ * r = p * s
+ * r = [0,0] if p = [0,0] or s mod q = 0
+ *
+ * @param[out]      r       Output point.
+ * @param           p       Input point.
+ * @param           s       Scalar.
+ *
+ * @retval -1 If r = [0,0].
+ * @retval 0  If 0 < s < q.
```
Comment (torsteingrindvik) [inline crypto/nrf_oberon/include/ocrypto_curve_p224.h:78]: It's unclear what `s == q` would return

File: crypto/nrf_oberon/include/ocrypto_curve_p224.h [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+/** P224 scalar base multiplication.
+ *
+ * r = basePoint * s
+ * r = [0,0] if s mod q = 0
+ *
+ * @param[out]      r       Output point.
+ * @param           s       Scalar.
+ *
+ * @retval -1 If r = [0,0].
+ * @retval 0  If 0 < s < q.
```
Comment (torsteingrindvik) [inline crypto/nrf_oberon/include/ocrypto_curve_p224.h:92]: It's unclear what s == q would return

File: crypto/nrf_oberon/include/ocrypto_curve_p224.h [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+ * p == [0,0] -> r = q
+ * q == [0,0] -> r = p
+ * p == -q    -> r = [0,0]
+ *
+ * @param[out]  r       Resulting point
+ * @param       p       Input point.
+ * @param       q       input point.
+ *
+ * @retval -1 if r = [0,0].
+ * @retval 0 if successfull.
```
Comment (torsteingrindvik) [inline crypto/nrf_oberon/include/ocrypto_curve_p224.h:109]: successfull -> successful

File: crypto/nrf_oberon/include/ocrypto_ecdh_p256.h [hunk 7]
```diff
@@ -55,10 +64,106 @@ int ocrypto_ecdh_p256_public_key(uint8_t r[64], const uint8_t s[32]);
...
+ * @code
+ *   ocrypto_ecdh_p256_public_key_init(ctx, sKey);
+ *   while (ocrypto_ecdh_p256_public_key_iterate(ctx));
+ *   res = ocrypto_ecdh_p256_public_key_final(ctx, pKey);
+ * @endcode
+ * Common Secret:
+ * @code
+ *   ocrypto_ecdh_p256_common_secret_init(ctx, sKey, pKey);
+ *   while (ocrypto_ecdh_p256_common_secret_iterate(ctx));
+ *   res = ocrypto_ecdh_p256_common_secret_final(ctx, secet);
```
Comment (torsteingrindvik) [inline crypto/nrf_oberon/include/ocrypto_ecdh_p256.h:85]: secet -> secret

**Review by torsteingrindvik — APPROVED** [review]

File: crypto/nrf_oberon/include/ocrypto_ecdh_p256.h [hunk 8]
```diff
@@ -55,10 +64,106 @@ int ocrypto_ecdh_p256_public_key(uint8_t r[64], const uint8_t s[32]);
 int ocrypto_ecdh_p256_common_secret(uint8_t r[32], const uint8_t s[32], const uint8_t p[64]);
 
 
+/**@name Incremental ECDH P-256 calculation.
```
Comment (b-gent) [inline crypto/nrf_oberon/include/ocrypto_ecdh_p256.h:67]:
```suggestion
/**@name Incremental ECDH P-256 calculation
```

**Review by b-gent — APPROVED** [review]
Adding comments with some nits that could be fixed, unless this is urgent to get in.
The doxygen groups are included properly in the build.

File: crypto/nrf_oberon/include/ocrypto_curve_p224.h (hunk 4 above)
Comment (frkv) [reply crypto/nrf_oberon/include/ocrypto_curve_p224.h:78]: -1 r = [0,0]

**Review by tejlmand — APPROVED** [review]
checked for no symbols

**Review by magnev — APPROVED** [review]

**Pull request merged**